File: LeetCode/problems/7-reverse-integer/7-reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let blah = 0;
    let sign = Math.sign(x);
    x = Math.abs(x);
    while(x > 0){
        blah = blah*10 + x%10;
        x = Math.floor(x/10);
    }
    if(blah > (2**31) - 1){ return 0; }
    return sign*blah;
};

File: LeetCode/problems/7-reverse-integer/7-reverse-integer.java
class Solution {
    public int reverse(int x) {
        String str = Long.toString(Math.abs((long)x));
        StringBuffer sbr = new StringBuffer(str);
        sbr.reverse();
        str = sbr.toString();
        if(x<0){str = "-"+str;}
        
        long value = Long.parseLong(str);
        int value2 = (int)Long.parseLong(str);
        if(value != value2){
            return 0;
        }
        return (int)value;
    }
}